lib/ecmascript.ts

```diff
@@ -1699,8 +1688,7 @@ export function CreateTemporalDate(
   isoDay: number,
   calendar: Temporal.CalendarProtocol = GetISO8601Calendar()
 ) {
-  const TemporalPlainDate = GetIntrinsic('%Temporal.PlainDate%');
-  const result = ObjectCreate(TemporalPlainDate.prototype);
+  const result = ObjectCreate(PlainDate.prototype);
```

Comment:
I'm having a difficult time understanding what is happening here - why is some `PlainDate` constructed here in this convoluted way? What exactly should `CreateTemporalDate` do?

What's the difference between
```js
const date = CreateTemporalDate(2021, 1, 1, 'iso8601')
```
and
```js
const date = new PlainDate(2021, 1, 1, 'iso8601')
```
?

Reply:
It's a really subtle difference — CreateTemporalDate is an internal operation that creates the object without doing any operations that could be observed from user code. `new PlainDate()` supports the case where the calendar argument is a property bag of the form of any other Temporal type that has a calendar, and accessing the `calendar` property of that property bag is potentially observable. For example, you could do
```js
const loggingDatePropertyBag = {
  year: 2021,
  month: 1,
  day: 1,
  get calendar() {
    console.log('calendar access observed!');
    return 'iso8601';
  },
};
const date = new Temporal.PlainDate(2021, 1, 1, loggingDatePropertyBag);
```
and this would log the "observed!" message.
CreateTemporalDate does not support this  usage. Whenever we call it it's always assumed that we have either already performed the observable operation, or ensured that it isn't necessary.

More info in the commit message of https://github.com/tc39/proposal-temporal/pull/1484/commits/12dab2e53f16a20da824408b0e922713c159c257